### ISSUE TITLE
1.17 update?

### ISSUE THREAD

**Issue by comonier** [PyvesB/AntiWorldFly#24]
waiting

**Comment by Sidpatchy** [comment]
As far as I'm aware, Spigot hasn't changed any aspect of the API that this plugin uses... I did testing right when Spigot 1.17 was released and didn't find any issues. It's possible I missed something, if so let me know because as it currently sits, I have no information on if there is an actual issue.

The issue templates exist for a reason, please use them (instead of deleting them) in the future.

**Comment by PyvesB** [comment]
> waiting

@comonier who do you think you are?

> It's possible I missed something, if so let me know because as it currently sits, I have no information on if there is an actual issue.

@Sidpatchy the clickable and hoverable messages that are used in the help won't work. You'll need to update one of the classes in MCShared. Nowadays I've pulled the functionality back into the last plugin I actively maintain: https://github.com/PyvesB/advanced-achievements/blob/master/advanced-achievements-plugin/src/main/java/com/hm/achievement/utils/FancyMessageSender.java

It's dreadful code, but it works and what can you do when you need to do heavy reflection across so many Minecraft versions... :)

**Comment by Sidpatchy** [comment]
That accidentally took way more effort than it needed to... whoops. At least I changed Sidpatchy/MCShared in a way that is beneficial in the long run :)

**Comment by PyvesB** [comment]
You used bungeecord-api, I've seen this around a few times, but I'm unclear about some of the details:
* is it available on non-Bungeecord servers?
* is it available on forks of Spigot such as Paper?
* is it available on older Minecraft versions, for example plain Bukkit 1.7.9?

Basically I've always wondered whether it would impact any of the existing users of the plugin.

**Comment by Sidpatchy** [comment]
> is it available on non-Bungeecord servers?

Yes! The bungeecord-chat api has been included in all **Spigot** servers since 1.8.

> is it available on forks of Spigot such as Paper?

Yes, Paper, Tuinity, and Purpur include the bungeecord-chat api as it is required to fully support Bungeecord.

> is it available on older Minecraft versions, for example plain Bukkit 1.7.9?

For versions prior to Spigot 1.8, no. I'm currently testing a few methods of bringing support CraftBukkit 1.7.9, however, versions after that will require Spigot or one of its forks.

TL;DR
It supports every version of Spigot since 1.8. It doesn't support any CraftBukkit versions (yet).
